packages/native-stack/src/utils/prepareHeaderBarButtonItems.ts

```diff
@@ -0,0 +1,82 @@
+import { Image, processColor } from 'react-native';
+
+import type {
+  NativeStackHeaderBarButtonItem,
+  NativeStackHeaderBarButtonItemWithCustomView,
+  NativeStackHeaderBarButtonItemWithMenu,
+} from '../types';
+
+const prepareMenu = (
+  menu: NativeStackHeaderBarButtonItemWithMenu['menu'],
+  index: number,
+  routeKey: string,
+  side: 'left' | 'right'
+): any => {
+  return {
+    ...menu,
+    items: menu.items.map((menuItem, menuIndex) => {
+      if ('items' in menuItem) {
+        return prepareMenu(menuItem, menuIndex, routeKey, side);
+      }
+      return {
+        ...menuItem,
+        menuId: `${menuIndex}-${index}-${routeKey}-${side}`,
```

Comment:
Since this just uses route key and indices for the id, can't this be done in React Native screens instead of React Navigation? We already pass some unique id to each screen that can be used.